controllers/topics/getLatestPost.js

```diff
@@ -64,12 +66,32 @@ module.exports = async (req, res) => {
 
     let message = `There are new posts from ${authorUsername}`;
 
+    const getstreamClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+    let unreadCount = 0;
+    try {
+      await getstreamClient.connectUser({id: userId}, token);
+
+      const userChannel = getstreamClient.channel('topics', `topic_${name}`);
+      const createdChannel = await userChannel.watch();
+      const readUsers = createdChannel?.read;
+      const selfUserReadStatus = readUsers?.find((readStatus) => {
+        return readStatus?.user?.id === userId;
+      });
+
+      unreadCount = selfUserReadStatus?.unread_messages || 0;
+    } catch (e) {
+      console.error('Error getting unread count', e);
+    } finally {
+      getstreamClient.disconnectUser();
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code connects to the StreamChat service, fetches the unread message count, and disconnects from the service. However, there's no error handling for the case where the connection to the StreamChat service fails. If the connection fails, the server will throw an unhandled promise rejection. To prevent this, consider adding error handling for the connection process.

Also, the API key and secret are directly accessed from the environment variables. It would be better to validate these values before using them. If they are undefined or invalid, the application should not proceed with the connection attempt.

```diff
-    const getstreamClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+    const apiKey = process.env.API_KEY;
+    const secret = process.env.SECRET;
+
+    if (!apiKey || !secret) {
+      console.error('API key or secret not provided');
+      return res.status(500).json({
+        status: 'error',
+        code: 500,
+        message: 'Internal Server Error'
+      });
+    }
+
+    const getstreamClient = new StreamChat(apiKey, secret);

     let unreadCount = 0;
     try {
       await getstreamClient.connectUser({id: userId}, token);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->